adapters/platform_linux.c

```diff
@@ -26,11 +27,19 @@
 
 int platform_init(void)
 {
-    int result;
+    int result = 0;
```

Comment:
>= 0 [](start = 15, length = 3)

We should not initialize default values.

You can do following :


int platform_init(void)
{
    int result;
#ifndef DONT_USE_UPLOADTOBLOB
    if (HTTPAPIEX_Init() == HTTPAPIEX_ERROR)
    {
        LogError("HTTP for upload to blob failed on initialization.");
        result = MU_FAILURE;
    }
    else
#endif /*DONT_USE_UPLOADTOBLOB*/
#ifdef USE_OPENSSL
    {
        result = tlsio_openssl_init();
    }
#else
    {
          result = 0;
    }
#endif

    return result;
}

Reply:
I agree, but, there will be the case where both code, UPLOADTOBLOB and OPENSSL, will be removed. In this case, the result will not be initialized. So, to avoid any problem, I'll take an exception here and pre-initialize the result.

---
In reply to: [319615071](https://github.com/Azure/azure-c-shared-utility/pull/366#discussion_r319615071) [](ancestors = 319615071)